packages/react/src/components/Tabs/TabItem/useTabItem.ts

```diff
@@ -0,0 +1,39 @@
+import { useContext, useId } from 'react';
+
+import { TabsContext } from '../Tabs';
+import type { ButtonProps } from '../../Button';
+
+import type { TabItemProps } from './TabItem';
+
+type UseTabItem = (props: TabItemProps) => {
+  active: boolean;
+  size?: 'small' | 'medium' | 'large';
+  buttonProps?: Pick<
+    ButtonProps,
+    'id' | 'onClick' | 'aria-checked' | 'aria-current'
+  >;
+};
+
+/** Handles props for `ToggleGroup.Item` in context with `ToggleGroup` and `RovingTabIndex` */
+export const useTabItem: UseTabItem = (props: TabItemProps) => {
+  const { ...rest } = props;
+  const tabs = useContext(TabsContext);
+  const itemValue =
+    props.value ?? (typeof props.children === 'string' ? props.children : '');
+  const active = tabs.value == itemValue;
+  const buttonId = `togglegroup-item-${useId()}`;
+
+  return {
+    ...rest,
+    active: active,
+    size: tabs?.size,
+    buttonProps: {
```

Comment:
Do you need the `buttonProps` here? Its only used on one level where its used which is a button, so maybe try to keep it simpler.